libavcodec/qsvenc_hevc.c

```diff
@@ -242,6 +242,7 @@ static const AVOption options[] = {
     { "mainsp",  NULL, 0, AV_OPT_TYPE_CONST, { .i64 = MFX_PROFILE_HEVC_MAINSP  }, INT_MIN, INT_MAX,     VE, "profile" },
     { "rext",    NULL, 0, AV_OPT_TYPE_CONST, { .i64 = MFX_PROFILE_HEVC_REXT    }, INT_MIN, INT_MAX,     VE, "profile" },
 
+    { "main10sp", "This profile allow to encode 10 bit single still picture", OFFSET(qsv.main10sp), AV_OPT_TYPE_BOOL, { .i64 = 0 }, 0, 1, VE},
```

Comment:
why it's a bool? not  a profile like mainsp?

Reply:
Main10sp is not an enum of profile (there is no enumeration like MFX_PROFILE_HEVC_MAIN10SP), so it cannot be added to "profile" option like mainsp.

Reply:
do we have the main still picture or the main 12/16 still picture?

Reply:
we have mainsp but don't have main 12/16 sp.
HM also process main10sp in this way - using main10 profile and enable one_pic_only flag.
According to specs, main10sp should set general_profile_idc to 2, which equals to main10. I think this is why they don't set a profile enumeration for main10sp.